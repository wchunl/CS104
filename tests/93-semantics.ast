ROOT "" (0.0.0)
|   ';' ";" (4.7.14)
|   TOK_FUNCTION "" (4.9.1)
|   |   TOK_TYPE_ID "" (4.9.1)
|   |   |   TOK_INT "int" (4.9.1)
|   |   |   TOK_IDENT "main" (4.9.5)
|   |   TOK_PARAM "(" (4.9.9)
|   |   TOK_BLOCK "{" (4.9.12)
|   |   |   TOK_VARDECL "=" (4.10.10)
|   |   |   |   TOK_INT "int" (4.10.4)
|   |   |   |   TOK_IDENT "c" (4.10.8)
|   |   |   |   '+' "+" (4.10.14)
|   |   |   |   |   TOK_IDENT "a" (4.10.12)
|   |   |   |   |   TOK_IDENT "b" (4.10.16)
|   |   |   TOK_VARDECL "=" (4.11.11)
|   |   |   |   TOK_VOID "void" (4.11.4)
|   |   |   |   TOK_IDENT "n" (4.11.9)
|   |   |   |   TOK_IDENT "null" (4.11.13)
|   |   |   TOK_VARDECL "=" (4.12.10)
|   |   |   |   TOK_INT "int" (4.12.4)
|   |   |   |   TOK_IDENT "x" (4.12.8)
|   |   |   |   '<' "<" (4.12.14)
|   |   |   |   |   TOK_IDENT "a" (4.12.12)
|   |   |   |   |   TOK_IDENT "b" (4.12.16)
|   |   |   TOK_VARDECL "=" (4.13.10)
|   |   |   |   TOK_INT "int" (4.13.4)
|   |   |   |   TOK_IDENT "y" (4.13.8)
|   |   |   |   TOK_EQ "==" (4.13.13)
|   |   |   |   |   TOK_IDENT "a" (4.13.12)
|   |   |   |   |   TOK_IDENT "b" (4.13.15)
|   |   |   TOK_RETURN "return" (4.14.4)
|   |   |   |   TOK_STRINGCON ""foobar"" (4.14.11)
