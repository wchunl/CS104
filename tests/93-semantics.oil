main:     .function int
          .local int c
          .local ptr n
          .local int x
          .local int y
          $t0:i = a + b
          c = $t0:i
          n = null
          $t1:i = a < b
          x = $t1:i
          $t2:i = a == b
          y = $t2:i
          return "foobar"
          return 
          .end
