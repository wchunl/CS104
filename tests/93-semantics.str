string_set[   3]:   11868120863286193964 0x1e1bdd8->"main"
string_set[   4]:   10838281452030117757 0x1e1ba78->"b"
                      21606623943142546 0x1e1bcc8->"{"
string_set[   6]:   13686721134187162920 0x1e1bd18->"}"
string_set[   8]:    1370730858159036685 0x1e1c118->"+"
string_set[   9]:   18322765439355934677 0x1e1cbb8->""foobar""
                   13272411544345499535 0x1e1c208->"x"
string_set[  12]:   15816506932648033945 0x1e1bc28->"("
string_set[  19]:   10959529184379665549 0x1e1bfb8->"c"
string_set[  21]:    8968846175329310707 0x1e1bb18->"f"
                   13581035385482130682 0x1e17608->"["
string_set[  23]:   12051031859636400188 0x1e1b8b8->"]"
                   12868445110721718657 0x1e1bc78->")"
string_set[  25]:    4993892634952068459 0x1e1b908->"a"
string_set[  28]:   14635384904416252381 0x1e1bbb8->"return"
string_set[  31]:   14736352883882276958 0x1e1c928->"=="
                    6142509188972423790 0x1e175e8->""
string_set[  32]:   11597697714117577063 0x1e1b988->"="
string_set[  34]:   15160494916537613830 0x1e1c468->"y"
string_set[  38]:   12597418120160561823 0x1e17218->"int"
string_set[  40]:   11911796782241914524 0x1e1bac8->"void"
string_set[  42]:    5344662657899890615 0x1e1ba28->";"
string_set[  43]:    9381151195890144283 0x1e1b9d8->"null"
string_set[  45]:    9826907452181669771 0x1e1c308->"n"
                    1396094175042499165 0x1e1c638->"<"
load_factor = 0.532
bucket_count = 47
max_bucket_size = 2
