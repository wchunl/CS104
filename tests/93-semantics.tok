#  1 "93-semantics.oc"
#  1 "<built-in>"
#  1 "<command-line>"
#  1 "93-semantics.oc"
   4 4.00001 262 TOK_INT       int
   4 4.00004  91 '['           [
   4 4.00005  93 ']'           ]
   4 4.00007 282 TOK_IDENT     a
   4 4.00009  61 '='           =
   4 4.00011 282 TOK_IDENT     null
   4 4.00015  59 ';'           ;
   4 5.00001 262 TOK_INT       int
   4 5.00004  91 '['           [
   4 5.00005  93 ']'           ]
   4 5.00007 282 TOK_IDENT     b
   4 5.00009  61 '='           =
   4 5.00011 282 TOK_IDENT     null
   4 5.00015  59 ';'           ;
   4 7.00001 261 TOK_VOID      void
   4 7.00005  91 '['           [
   4 7.00006  93 ']'           ]
   4 7.00008 282 TOK_IDENT     f
   4 7.00009  40 '('           (
   4 7.00010  41 ')'           )
   4 7.00012 123 '{'           {
   4 7.00013 125 '}'           }
   4 7.00014  59 ';'           ;
   4 9.00001 262 TOK_INT       int
   4 9.00005 282 TOK_IDENT     main
   4 9.00009  40 '('           (
   4 9.00010  41 ')'           )
   4 9.00012 123 '{'           {
   4 10.00004 262 TOK_INT       int
   4 10.00008 282 TOK_IDENT     c
   4 10.00010  61 '='           =
   4 10.00012 282 TOK_IDENT     a
   4 10.00014  43 '+'           +
   4 10.00016 282 TOK_IDENT     b
   4 10.00017  59 ';'           ;
   4 11.00004 261 TOK_VOID      void
   4 11.00009 282 TOK_IDENT     n
   4 11.00011  61 '='           =
   4 11.00013 282 TOK_IDENT     null
   4 11.00017  59 ';'           ;
   4 12.00004 262 TOK_INT       int
   4 12.00008 282 TOK_IDENT     x
   4 12.00010  61 '='           =
   4 12.00012 282 TOK_IDENT     a
   4 12.00014  60 '<'           <
   4 12.00016 282 TOK_IDENT     b
   4 12.00017  59 ';'           ;
   4 13.00004 262 TOK_INT       int
   4 13.00008 282 TOK_IDENT     y
   4 13.00010  61 '='           =
   4 13.00012 282 TOK_IDENT     a
   4 13.00013 275 TOK_EQ        ==
   4 13.00015 282 TOK_IDENT     b
   4 13.00016  59 ';'           ;
   4 14.00004 268 TOK_RETURN    return
   4 14.00011 285 TOK_STRINGCON "foobar"
   4 14.00019  59 ';'           ;
   4 15.00001 125 '}'           }
