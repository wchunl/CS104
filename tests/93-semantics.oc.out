; 0.  1: # 1 "93-semantics.oc"; 1.  0: 
; 1.  1: # 1 "<built-in>"
; 2.  1: # 1 "<command-line>"
; 3.  1: # 1 "93-semantics.oc"
; 4.  1: 
; 4.  2: 
; 4.  3: 
; 4.  4: int[] a = null;
; 4.  5: int[] b = null;
; 4.  6: 
; 4.  7: void[] f() {};
; 4.  8: 
; 4.  9: int main() {
; 4. 10:    int c = a + b;
; 4. 11:    void n = null;
; 4. 12:    int x = a < b;
; 4. 13:    int y = a==b;
; 4. 14:    return "foobar";
; 4. 15: }
hello